### PR TITLE
scripts/create_addon: add changelog to new "news" item

### PR DESCRIPTION
**Pull request by lrusak**
Simple enough, if someone has a better solution please let me know. I had a hard time figuring out the \n line endings in order for the sed command to work.

this changed in, https://github.com/xbmc/xbmc/pull/9351

### CODE REVIEW

**Comment by MilhouseVH** [comment]
The problem with this is that when `$PKG_DIR/changelog.txt` includes a pipe symbol, the `sed` substitution will fail.

What will work reliably is the following (add it after the main sed in line 105 and then drop the `while...done` lines 81..83 as these are no longer required):
```
if [ -f $PKG_DIR/changelog.txt ];
  sed -e "/@PKG_ADDON_NEWS@/ \
          {
            r $PKG_DIR/changelog.txt
            d
          }" -i $ADDON_BUILD/$PKG_ADDON_ID/addon.xml
else
  sed -e "s|@PKG_ADDON_NEWS@||g" -i $ADDON_BUILD/$PKG_ADDON_ID/addon.xml
fi
```

The only issue is when `changelog.txt` doesn't have a carriage return on the last line, as you'll end up with:
```
    <news>
first line
last line    </news>
```
but this is probably just a cosmetic issue.

**Comment by lrusak** [comment]
@MilhouseVH that doesn't make sense to me, where does the replacement text come from?

**Comment by MilhouseVH** [comment]
`r $PKG_DIR/changelog.txt` - this reads the contents of the file.
`d` deletes the pattern space.

**Comment by lrusak** [comment]
Ah of course.

File: scripts/create_addon [hunk 1]
```diff
@@ -74,8 +74,13 @@ pack_addon() {
     cp $PKG_DIR/icon/icon.png $ADDON_BUILD/$PKG_ADDON_ID
   fi
 
+  PKG_ADDON_NEWS=""
   if [ -f $PKG_DIR/changelog.txt ]; then
     cp $PKG_DIR/changelog.txt $ADDON_BUILD/$PKG_ADDON_ID
+
+    while read -r LINE; do
+      PKG_ADDON_NEWS+="$LINE\n"
+    done < $PKG_DIR/changelog.txt
```
Comment (awiouy) [inline scripts/create_addon]: PKG_ADDON_NEWS="$(cat $PKG_DIR/changelog.txt)"

**Comment by MilhouseVH** [comment]
I guess `changelog.txt` files created on Windows will have LF CR formatting, does this cause problems for Kodi on Linux when displaying the `news` (ie. will it be double spaced text etc.)?

Perhaps pre-processing `changelog.txt` to remove any possible carriage returns, and even ensuring it has a trailing line feed, would be advisable:

```
if [ -f $PKG_DIR/changelog.txt ];
  cat $PKG_DIR/changelog.txt | tr -d '\r' | sed '$a\' >/tmp/changelog.txt
  sed -e "/@PKG_ADDON_NEWS@/ \
          {
            r /tmp/changelog.txt
            d
          }" -i $ADDON_BUILD/$PKG_ADDON_ID/addon.xml
  rm /tmp/changelog.txt
else
  sed -e "s|@PKG_ADDON_NEWS@||g" -i $ADDON_BUILD/$PKG_ADDON_ID/addon.xml
fi
```

`tr -d '\r'` will remove any carriage returns
`sed '$a\'` will add a linefeed at the end of the file only if the last line is missing the linefeed

**Comment by lrusak** [comment]
updated. 

I left out the pre-processing because we provide the changelog.txt files and addon.xml files so they should be correct and if they aren't we can fix them.

**Comment by MilhouseVH** [comment]
> I left out the pre-processing because we provide the changelog.txt files
Fair enough, thanks.

File: scripts/create_addon [hunk 2]
```diff
@@ -96,6 +102,7 @@ pack_addon() {
        -e "s|@PKG_DISCLAIMER@|$PKG_DISCLAIMER|g" \
        -e "s|@PROVIDER_NAME@|$PROVIDER_NAME|g" \
        -e "s|@PKG_ADDON_PROVIDES@|$PKG_ADDON_PROVIDES|g" \
+       -e "s|@PKG_ADDON_NEWS@|$PKG_ADDON_NEWS|g" \
```
Comment (MilhouseVH) [inline scripts/create_addon]: ^^ No longer needed

**Comment by lrusak** [comment]
updated. thanks for catching that.

**Comment by MartijnKaijser** [comment]
Sorry but I find this just wrong. News was supposed to be a short changelog and not a complete dump again.

**Pull request merged**